modules/application/src/main/java/edu/northeastern/truthtree/service/basicInfo/BasicInfoService.java

```diff
@@ -37,13 +37,13 @@ public BasicInfoService(IBasicInfoAdapter adapter) {
    * @return JSONArray that contains states that are within the provided range.
    */
   @Override
-  public JSONArray getBasicStatesInfo(int[] range) {
-
+  public PageDTO<LocationDTO> getBasicStatesInfo(int page, int[] range)
+          throws IOException, IllegalArgumentException {
     if (range != null) {
       if (range.length == 2 && range[0] <= range[1]) {
-        return this.adapter.getBasicStatesInfo(range[0], range[1]);
+        return this.adapter.getBasicStatesInfo(page, range[0], range[1]);
       }
-      return JSONUtil.createErrorMessage(POPULATION_ERROR);
+      throw new IllegalArgumentException("Invalid range");
```

Comment:
Can you add error message in ErrorMessages class or reuse the same message?

Reply:
addressed